src/api/spec/models/workflow/step/submit_request_step_spec.rb

```diff
@@ -31,12 +31,15 @@
       target_project: target_project.name
     }
   end
+  let(:scm_status_reporter) { instance_double(SCMStatusReporter) }
   let(:workflow_run) { create(:workflow_run, token: token) }
 
   describe '#call' do
     before do
       login(user)
       allow(Backend::Api::Sources::Package).to receive(:wait_service).and_return(true)
+      allow(SCMStatusReporter).to receive(:new).and_return(scm_status_reporter)
```

Comment:
This change is required because we want to test the `Workflow::Step::SubmitRequest` model with successful calls to `SCMStatusReporter`.

Before, we rescued from an exception on an unauthorized request to an SCM on an upper level, the `Token::Workflow` class.

Now, we rescue from an exception on an unauthorized request to an SCM just after performing the request to the SCM. That is why we have to mock a successful call to `SCMStatusReporter`. Not mocking successfull calls to `SCMStatusReporter` will result in exceptions we don't want to deal within `Workflow::Step::SubmitRequest` tests.

For example, with the GitHub SCM case, we moved the rescue code into the `GithubStatusReporter` class. This class is used by `SCMStatusReporter`. And `SCMStatusReporter` is used by `Workflow::Step::SubmitRequest`.

Reply:
alright, thanks for explaining :)